test/App/Fossa/ManualDepsSpec.hs

```diff
@@ -77,6 +154,28 @@ spec = do
         Left err -> expectationFailure err
         Right jsonDeps -> jsonDeps `shouldBe` theWorks
 
+  describe "fossa-deps labeled" $ do
+    theWorksLabeledBS <- getTestDataFile "the-works-labeled.yml"
+    it "should successfully parse all possible inputs" $
+      case Yaml.decodeEither' theWorksLabeledBS of
+        Left err -> expectationFailure $ displayException err
+        Right yamlDeps -> yamlDeps `shouldBe` theWorksLabeled
+
+    it "should extract nested labels with orgId" $
+      case Yaml.decodeEither' theWorksLabeledBS of
+        Left err -> expectationFailure $ displayException err
+        Right yamlDeps -> do
+          let orgId = OrgId 1234
+          let extracted = collectInteriorLabels (Just orgId) yamlDeps
+          extracted `shouldBe` theWorksLabels (Just orgId)
+
+    it "should extract nested labels without orgId" $
+      case Yaml.decodeEither' theWorksLabeledBS of
+        Left err -> expectationFailure $ displayException err
+        Right yamlDeps -> do
+          let extracted = collectInteriorLabels Nothing yamlDeps
+          extracted `shouldBe` theWorksLabels Nothing
```

Comment:
[Q] Would it be worth round-tripping the `FromJSON/ToJSON` instances?

Reply:
We can't round trip them: -- | Note: The ToJSON and FromJSON implementations for this type differ
-- because ToJSON is used to upload to Core and FromJSON is used to parse
-- from a fossa-deps file.

Reply:
If we really want to I could of course split them into two types for this, which would then let us round trip, but I didn't think it was worth the effort in this case especially given our future plans around moving towards an observation model. 

If you disagree happy to reconsider!

Reply:
Ah no that sounds good. Sorry I missed that!